client/component/text.js

```diff
@@ -238,13 +238,25 @@ export function AutoSizeTextInput({value, onChange, placeholder, style, hoverSty
     </View>
 }
 
+const FocusContext = createContext();
+export const useFocus = () => useContext(FocusContext);
+
+export const FocusProvider = ({ children }) => {
```

Comment:
Rather than having a context here, can we just have an optional onFocusChange callback on the TextField?

TextField is a direct child of EditComment, so we don't need to worry about flowing state through a deep tree of components (which is what contexts are useful for).

Also, in the code base we are largely avoiding use of custom contexts, because we can manage most of that more easily using sessionData (useSessionData and datastore.setSessionData).

Reply:
Yes. thanks for catching. I don't know how I missed that TextField was a direct child... somehow I had convinced myself the tree had many components in between.